tasks/fullStack/spec/services/tag_service_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+RSpec.describe TagService do
+  describe ".tags_with_default" do
+    let(:product) { create(:product) }
+    let(:shop) { product.shop }
+
+    context "when the product does not exist" do
+      it "returns the given and default tags" do
+        result = TagService.tags_with_default(-1, "given_tag1,given_tag2")
+
+        expect(result).to match_array(%w[default given_tag1 given_tag2])
+      end
+    end
+
+    context "when the product exists" do
+      context "when the shop has tags" do
+        before do
+          shop.update(tags: %w[shop_tag1 shop_tag2])
+        end
```

Comment:
Can you please explain why you are using `before` block to update `shop` record instead of `let(:shop)`? Is it a personal preference or is there a reason for it?

```suggestion
      let(:shop) { create(:shop, tags: %w[shop_tag1 shop_tag2]) }
```

Reply:
I wish I could give a reason, but I don't know why I did it like that 😅 `let(:shop) { create(:shop, tags: %w[shop_tag1 shop_tag2]) }` would have been cleaner for sure.